### PR TITLE
EQL: Extract query folder tests definitions into resources

### PR DESCRIPTION
**Pull request by aleksmaus**
@costin, this addresses the other ask that you had for extracting the tests specs into resources.
As far as I understood we were talking about simple text file, but can change the format however needed. Let me know.

Screen Shot 2020-03-19 at 10 24 39 AM

### CODE REVIEW

File: x-pack/plugin/eql/src/test/resources/queryfolder_tests.txt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+
+basic
+process where true
+null
+
+
+singleNumericFilterEquals
+process where serial_event_id = 1
+"term":{"serial_event_id":{"value":1
```
Comment (imotov) [inline x-pack/plugin/eql/src/test/resources/queryfolder_tests.txt]: These unclosed parentheses are tearing holes in my OCD soul. :smile:

**Review by imotov — APPROVED** [review]
I really like where it is going, and I understand that this is just a string comparison at the moment. However, I wonder if it is possible to make the testing expressions a bit more JSON-like and make it not rely on the order of fields in JSON objects. 

One idea would be to just put json parse it into a map and make sure this it is "submap" of the response. So basically something like `{"foo": {"baz":2"}}` would match response `{"foo": {"bar": 1, "baz":2"}}`.

**Comment by aleksmaus** [comment]
@imotov that how it was done originally in the previous PR, I loaded into maps and compared maps, so didn't have to rely on the order, but after discussion with @costin changed that back to substrings match.

**Comment by imotov** [comment]
What's the motivation?

**Comment by aleksmaus** [comment]
@imotov 
1. comparing the full maps makes the test too fragile dependent on other internal ES implementation changes
2. traversing maps tree and match submaps makes the definition of the tests specs a bit more complex
so the decision was the substrings match was adequate, sufficient and less fragile.
@costin might chime in more on pros on cons.

**Review by costin — APPROVED** [review]
LGTM

**Comment by aleksmaus** [comment]
Rebased against the latest master to resolve the conflicts and force pushed.

**Comment by costin** [comment]
@aleksmaus Use merging instead of rebase to preserve history.

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderOkTests.java [hunk 2]
```diff
@@ -7,72 +7,19 @@
...
-            {"equalsAndInFilter", "process where process_path == \"*\\\\red_ttp\\\\wininit.*\" and opcode in (0,1,2,3)",
-                    new Object[]{
-                            "\"wildcard\":{\"process_path\":{\"wildcard\":\"*\\\\\\\\red_ttp\\\\\\\\wininit.*\"",
-                            "\"term\":{\"opcode\":{\"value\":0",
-                            "\"term\":{\"opcode\":{\"value\":1",
-                            "\"term\":{\"opcode\":{\"value\":2",
-                            "\"term\":{\"opcode\":{\"value\":3",
-                    }
-            },
-            {"substringFunction", "process where substring(file_name, -4) == '.exe'",
```
Comment (costin) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderOkTests.java]: This test was skipped during externalization. Discovered that during backporting.

**Pull request merged**